mqtt/mqtt.js

```diff
@@ -80,14 +80,16 @@ function startMqtt() {
     return net.createServer(aedes.handle).listen(config.mqtt_port);
 }
 
-nats.subscribe('channel.*', {'queue':'mqtts'}, function (msg) {
+nats.subscribe('channel.>', {'queue':'mqtts'}, function (msg) {
     var m = message.RawMessage.decode(Buffer.from(msg)),
-        packet;
+        packet, channel;
     if (m && m.Protocol !== 'mqtt') {
+        channel = /^([\w\-]+)(\.(.+))?$/.exec(m.Channel)
         packet = {
             cmd: 'publish',
             qos: 2,
-            topic: 'channels/' + m.Channel + '/messages',
+            topic: 'channels/' + channel[1] + '/messages' + (
```

Comment:
Just curious, what is channel[0] ?

Reply:
hi,

0 is always all the string if match the regexp
1 channel id
2 the remaing part  of the string or undefined
3 the remaing of the string or undefined

as example is m.Channel is "a22de3db-61df-461a-806f-5c2ed7f4cb10.data" channel var will be
[ 'a22de3db-61df-461a-806f-5c2ed7f4cb10.data', 'a22de3db-61df-461a-806f-5c2ed7f4cb10',  '.data',  'data',  index: 0,  input: 'a22de3db-61df-461a-806f-5c2ed7f4cb10.data' ]

while if m.Channel is "a22de3db-61df-461a-806f-5c2ed7f4cb10" channel var will be
[ 'a22de3db-61df-461a-806f-5c2ed7f4cb10', 'a22de3db-61df-461a-806f-5c2ed7f4cb10',  undefined, undefined,  index: 0,  input: 'a22de3db-61df-461a-806f-5c2ed7f4cb10' ]

Reply:
Got it, thank you1